src/query/tsdb/remote/codecs.go

```diff
@@ -269,3 +269,10 @@ func decodeTagMatchers(rpcMatchers *rpc.TagMatchers) (models.Matchers, error) {
 
 	return models.Matchers(matchers), nil
 }
+
+// DecodeSearchRequest decodes rpc search request to read query and read options
+func DecodeSearchRequest(
+	req *rpc.SearchRequest,
+) (*storage.FetchQuery, error) {
+	return &storage.FetchQuery{}, nil
```

Comment:
nit: Can we declare `var errNotImplemented = errors.New("not implemented")` and `return errNotImplemented` here until implemented?  Then at least it is transparent about behavior.

Reply:
Sure, wasn't too worried about it since this PR should implement the function before the meat of it is up for review :)